src/main/java/com/synopsys/integration/detect/lifecycle/autonomous/AutonomousManager.java

```diff
@@ -144,10 +142,35 @@ private Optional<String> findScanType(List<String> adoptedScanTypes, String key)
         }).findFirst();
     }
 
+    private Optional<String> findDetectorType(List<String> detectorTypes, String key) {
+        return detectorTypes.stream().filter(detector -> {
+            String tempKey = key.replace(".","_");
+            return tempKey.contains(detector.toLowerCase());
+        }).findFirst();
+    }
+
+    private void updateScanOrDetectorProperty(String propertyType, Boolean userProvidedProperty, String propertyTypeName, String propertyKey, String propertyValue) {
+        if(propertyType.equals("Detector")) {
+            PackageManagerType packageManagerType = scanSettings.getDetectorTypeWithName(propertyTypeName);
+            if(userProvidedProperty && !propertyValue.isEmpty()) {
+                packageManagerType.getDetectorProperties().put(propertyKey, propertyValue);
+            } else if(!packageManagerType.getDetectorProperties().containsKey(propertyKey) && !propertyValue.isEmpty()) {
+                packageManagerType.getDetectorProperties().put(propertyKey, propertyValue);
+            }
```

Comment:
Looks like this can be a single if condition (can also be a boolean helper method if you like)